crates/sncast/src/lib.rs

```diff
@@ -664,7 +673,14 @@ pub async fn handle_wait_for_tx<T>(
     ui: &UI,
 ) -> Result<T, WaitForTransactionError> {
     if wait_config.wait {
-        return match wait_for_tx(provider, transaction_hash, wait_config.wait_params, ui).await {
+        return match wait_for_tx(
+            provider,
+            transaction_hash,
+            wait_config.wait_params,
+            if wait_config.silent { None } else { Some(ui) },
```

Comment:
nit:
```suggestion
            (!wait_config.silent).then(|| ui),
```

Reply:
Don't think that's more readable, especially having to call the method on `(!...)`